docs/content/preview/quick-start/build-apps/rust/ysql-diesel.md

```diff
@@ -0,0 +1,329 @@
+------------+-------------------+-------+--------------
+          1 | 200 page notebook |  7.50 | Notebook
+          2 | Mechanical pencil |  2.50 | Pencil
+(2 rows)
+```
+
+Create 2 orders using the `userId` for John.
+
+```sh
+$ curl \
+  --data '{ "userId": 1, "products": [ { "productId": 1, "units": 2 } ] }' \
+  -v -X POST -H 'Content-Type:application/json' http://localhost:8080/orders
+```
+
+```sh
+$ curl \
+  --data '{ "userId": 1, "products": [ { "productId": 1, "units": 2 }, { "productId": 2, "units": 4 } ] }' \
+  -v -X POST -H 'Content-Type:application/json' http://localhost:8080/orders
+```
+
+## Query results
+
+### Using the YSQL shell
+
+In your YSQL shell, list the tables created by the application.
+
+```sql
+ysql_diesel=#  \d
+```
+
+```output
+                      List of relations
+ Schema |             Name              |   Type   |  Owner   
+--------+-------------------------------+----------+----------
+ public | __diesel_schema_migrations    | table    | yugabyte
+ public | order_lines                   | table    | yugabyte
+ public | order_lines_order_id_seq      | sequence | yugabyte
+ public | order_lines_order_line_id_seq | sequence | yugabyte
+ public | order_lines_product_id_seq    | sequence | yugabyte
+ public | orders                        | table    | yugabyte
+ public | orders_order_id_seq           | sequence | yugabyte
+ public | orders_user_id_seq            | sequence | yugabyte
+ public | products                      | table    | yugabyte
+ public | products_product_id_seq       | sequence | yugabyte
+ public | users                         | table    | yugabyte
+ public | users_user_id_seq             | sequence | yugabyte
+(12 rows)
+```
+
+Note the 4 tables and 3 sequences in the list above.
+
+```sql
+ysql_diesel=# SELECT count(*) FROM users;
+```
+
+```output
+ count
+-------
+     2
+(1 row)
+```
+
+```sql
+ysql_diesel=# SELECT count(*) FROM products;
+```
+
+```output
+ count
+-------
+     2
+(1 row)
+```
+
+```sql
+ysql_diesel=# SELECT count(*) FROM orders;
+```
+
+```output
+ count
+-------
+     2
+(1 row)
+```
+
+```sql
+ysql_diesel=# SELECT * FROM order_lines;
+```
+
+```output
+ order_line_id | order_id | product_id | units 
+---------------+----------+------------+-------
+             1 |        1 |          1 |     2
+             2 |        2 |          1 |     2
+             3 |        2 |          2 |     4
+(3 rows)
+```
+
+`order_lines` is a child table of the parent `orders` table, and is connected using a foreign key constraint. The `users` table is connected with `orders` using a foreign key constraint so that no order can be placed with an invalid user, and that user has to be present in the users table.
+
+### Using the REST API
+
+To use the REST API server to verify that the users, products, and orders were created in the `ysql_diesel` database, enter the following commands. The results are output in JSON format.
+
+```sh
+$ curl http://localhost:8080/users
```

Comment:
I dont' think we should show the prompt in the commands (because the reader already has a prompt on his terminal and copy-paste will double it)